bin/uniref90_rhea_filter.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import gzip
+
+from Bio import SeqIO
+
+
+def load_mapping(tsv_file):
+    """
+    Load mapping of UniRef90 proteins to Rhea IDs from a TSV file.
+    Returns a dictionary with RepID as the key and RheaID as the value.
+    """
+    mapping = {}
+    with open(tsv_file, 'r') as file:
+        for line in file:
+            rep_id, rhea_id = line.strip().split('\t')
+            rhea_id = " ".join(sorted(set(rhea_id.split())))
+            mapping[rep_id] = rhea_id
+    return mapping
+
+
+def filter_fasta(in_handle, out_handle, mapping):
+    """
+    Filter the FASTA file based on the mapping and add RheaID to the header.
+    """
+    for record in SeqIO.parse(in_handle, "fasta"):
+        rep_id = record.id.split("_")[1]
```

Comment:
Maybe it's worth validating the fasta header here? In case something goes wrong, and the header doesn't have the right format

Reply:
Records in Uniref90 always looks like `UniRef90_XXXX` where XXXX is protein id and UniRef90_ means that this protein is from UniRef90 DB
If ID is different format, it means something went **_totally_** wrong on their site and I am unsure if we need to handle this case...

Ofc, I can add index error, but what to do in the exception block? Skip and continue?

Reply:
If something when totally here.. I would throw an exception

I would use a string comparison or regex to check that "UniRef90_XXXX" is here.. otherwise throw an exception and which will stop the pipeline

Reply:
Added in [raise exception if different protein name format](https://github.com/EBI-Metagenomics/taxdb_generation_nf/pull/1/commits/7203bbdf3b617f1cde0438529ea40d94429801d4)